src/Bicep.LangServer/Completions/PublicRegistryModuleMetadataProvider.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft;
+using Newtonsoft.Json;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+
+namespace Bicep.LanguageServer.Providers
+{
+    public record ModuleMetadata(string moduleName, List<string> tags);
+
+    /// <summary>
+    /// Provider to get modules metadata from this endpoint - https://live-data.bicep.azure.com/modulesMetadata
+    /// The above endpoint helps fetch module names and versions of modules available in this github repository - https://github.com/Azure/bicep-registry-modules
+    /// </summary>
+    public class PublicRegistryModuleMetadataProvider : IPublicRegistryModuleMetadataProvider
+    {
+        private const string LiveDataEndpoint = "https://live-data.bicep.azure.com/modulesMetadata";
+        private List<ModuleMetadata> moduleMetadataCache = new List<ModuleMetadata>();
+
+        public async Task Initialize()
```

Comment:
It might be worth a team discussion on how to handle the following failure modes:
* No internet connection (looks like you've already handled this with the try-catch)
* Slow or spotty internet connection
* Transient failure reaching the module metadata endpoint

My personal thoughts:
* We should not delay startup or any UI (e.g. requesting completions) if the user's internet connection is slow.
* We should have retry logic in case there is a transient failure.
* We may want to find an unobtrusive way to surface information about failures to explain what the user may percieve as "unreliable" behavior (e.g. if their internet connection drops out, and suddenly certain completions are unavailable).
* It may be worth implementing caching.

AFAIK, the module registry implementation handles all of the above - it may be worth converging on a solution.

Reply:
@anthony-c-martin, we(@puicchan, @StephenWeatherford and @majastrz) discussed this issue yesterday. One option was to put these changes behind a feature flag and create an issue to address the concerns that you mentioned.

I made changes to add new experimental flag:
https://github.com/Azure/bicep/commit/383543588f2e4b9a33bd6f74748f1a5e6cf1341f

And created issue and assigned it to Stephen:
https://github.com/Azure/bicep/issues/9975

Let me know what you think. Thanks!

Reply:
Sounds good to me.